compiler/ifa/num.cpp

```diff
@@ -183,26 +183,32 @@ snprint_imm(char *str, size_t max, Immediate &imm) {
 }
 
 int 
-fprint_imm(FILE *fp, Immediate &imm) {
+fprint_imm(FILE *fp, Immediate &imm, bool showType) {
   int res = -1;
   switch (imm.const_kind) {
     case NUM_KIND_NONE:
+      if (showType) res = fputs(":NONE", fp); break;
       break;
     case NUM_KIND_BOOL: {
       // Since I'm using uints to store the bools, should cast to the
       // same to be consistent.
       res = fprintf(fp, "%"PRIu64, imm.bool_value()); break;
+      if (showType) res += fprintf(fp, " :bool"); break;
```

Comment:
It is a little strange that this line uses fprintf when every other print added in this function uses fputs.